build/scripts/install_externals.py

```diff
@@ -0,0 +1,207 @@
+    if json_blob == None:
+        print("Loaded json blob is None, stopping json parse")
+        return []
+
+    externals_required = []
+    for object in json_blob:
+        externals_required.append(json_blob[object])
+
+    return externals_required
+
+def __download_external(url):
+    # get basename of the URL (a.k.a. the filename + extention we would like to download)
+    url_basename = os.path.basename(url)
+
+    # request a download of the given URL
+    if not os.path.exists(url_basename):
+        response = requests.get(url)
+        if response.status_code == requests.codes.ok:
+            # write the downloaded file to disk
+            open(url_basename, "wb").write(response.content)
+        else:
+            # bad request was made
+            print("Bad request [" + str(response.status_code) + "] for given url: " + url)
+            return 1
+        
+    # extract the zip file on disk
+    # we cache the files within the directory before 
+    # and after extraction, this gives us the ability
+    # to examine the added files within the directory
+    print("Extracting: " + url)
+    
+    # pre list directories
+    # cached directories before we downloaded anything
+    pre_list_dir = os.listdir(__get_script_path())
+    # print("pre-list-dir: " + " ".join(pre_list_dir))
+    with zipfile.ZipFile(url_basename,"r") as zip_ref:
+        zip_ref.extractall(__get_script_path())
+
+    # post list directories
+    # directories after we downloaded the repository
+    post_list_dir = os.listdir(__get_script_path())
+    # print("post-list-dir: " + " ".join(post_list_dir))
+
+    print("Looking for added extracted directories ...")
+    added_directory_names = []
+    for post_dir in post_list_dir:
+        count = pre_list_dir.count(post_dir)
+        if count == 0:
+            added_directory_names.append(post_dir)
+    print("Found (" + str(len(added_directory_names)) + "): " + " ".join(added_directory_names))
+
+    # remove the created zip file
+    os.remove(url_basename)
+
+    return added_directory_names
+
+def __touch_externals_dir(externalsDir, tag):
+    version = {
+        "tag": tag
+    }
+
+    json_object = json.dumps(version, indent=4)
+
+    # Writing to version.json
+    with open(os.path.join(externalsDir, "version.json"), "w") as out:
+        out.write(json_object)
+
+def __install_external(external):
+    cwd = os.getcwd()
+    root = __get_root_path()
+
+    external_url = external["url"]
+    external_name = external["name"]
+    external_tag = external["tag"]
+    external_store = external["storage"]
+    external_store = external_store.replace("~", root)
+
+    # if the external is already present we need to check if we need to redownload anything
+    should_download = False
+    externals_dir = os.path.join(external_store, external_name)
+    if os.path.exists(externals_dir):
+        print("External found: " + external_name + " validating version ...")
+        os.chdir(externals_dir)
```

Comment:
I don't think you have to change the directory, keep using the absolute paths, it avoids a missing error where you put the cwd back to what it was